collectors/osd.go

```diff
@@ -687,10 +738,29 @@ func (o *OSDCollector) collectOSDScrubState(ch chan<- prometheus.Metric) error {
 	return nil
 }
 
+func (o *OSDCollector) collectPGRecoveryState(ch chan<- prometheus.Metric) error {
+	for _, pg := range o.pgDumpBrief {
+		if strings.Contains(pg.State, "recovering") {
+			query, err := o.performPGQuery(pg.PGID)
+			if err != nil {
+				continue
+			}
+
+			ch <- prometheus.MustNewConstMetric(
+				o.PGObjectsRecoveredDesc,
+				prometheus.CounterValue,
+				float64(query.Info.Stats.StatSum.NumObjectsRecovered),
```

Comment:
Should we surface the delta of objects instead of actual number of objects recovered till date? If we plug this value in directly from Ceph then the counter will be increment the existing value by the total number of recovered objects. Since we want to track the rate I think we should keep track of the total number of objects recovered till date and then only add deltas here.

So for e.g. for following sequence
1. pg query (first)
2. num objects recovered: 5
3. pg query (next second)
4. num objects recovered: 5 

This means no change in objects recovered and rate should be 0. But with current change the rate will be 5.

Reply:
Yep, my intention is the same. But I was not aware that the value fed into the `MustNewConstMetric` for counter is the increment, but not the absolute value.